### PR TITLE
#158141259 Build out edge cases tests

### PR DESCRIPTION
**Pull request by denniswanjiru**
#### What does this PR do?
Build out edge cases to handle all the conditional logic in the app

#### Description of Task to be completed?
- Add tests to handle username uniqueness
- Add tests to handle email uniqueness
- Add tests to handle incorrect password
- Add tests to handle password confirmation  miss-match
- Add tests to handle requests owner

#### How should this be manually tested?
1. Clone the branch into your local machine as follows
2. Run the following commands on your terminal 

```
git clone -b ch-add-egdecases-tests-158141259 git@github.com:DennisWanjiru/maintenance-tracker.git 

cd maintenance-tracker/

python3 -m virtualenv env -p python3.6

source env/bin/activate

pip install -r requirements.txt

nosetests
```

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#158141259](https://www.pivotaltracker.com/story/show/158141259)

### CODE REVIEW

File: tests/test_auth.py [hunk 1]
```diff
@@ -69,6 +69,91 @@ def test_signout(self):
...
+                name="Dennis",
+                password="root",
+                confirm_password="rooter"
+            )),
+            content_type=("application/json")
+        )
+
+        self.assertEqual(response.status_code, 400)
+
+    def test_email_taken(self):
```
Comment (daisymacharia) [inline tests/test_auth.py]: You have already defined this test on line 88

File: tests/test_auth.py [hunk 2]
```diff
@@ -69,6 +69,91 @@ def test_signout(self):
...
+            '/api/v1/users/auth/signup/',
+            data=json.dumps(self.data["user"]),
+            content_type=("application/json")
+        )
+
+        self.assertEqual(response.status_code, 400)
+
+    def test_email_taken(self):
+        """ Test if the email is already taken """
+        self.client.post(
```
Comment (daisymacharia) [inline tests/test_auth.py:90]: for "DRY-er" code you could define the register as a function that you call whenever you require to make a post to `/api/v1/users/auth/signup/` ... unless in cases where you need to test like passwords not matching.

File: tests/test_requests_resource.py [hunk 3]
```diff
@@ -73,6 +98,42 @@ def test_get_a_request(self):
         response = self.client.get('/api/v1/users/request/1/')
         self.assertEqual(response.status_code, 200)
 
+        def test_get_a_non_existing_request(self):
+            """ Test user tries to retrieve a noin-existing request """
+            self.create_and_login_user()
+
+            response = self.client.post(
```
Comment (daisymacharia) [inline tests/test_requests_resource.py]: you can have this as a function that is called to create the request whenever you need to.
